### DISCUSSION TITLE
How to start weed volume when some dat or idx files were missing or damaged.

### DISCUSSION THREAD

**Discussion by guol-fnst** [seaweedfs/seaweedfs#2572] [Q&A] [answered]
How to start weed volume when some dat or idx files were missing or damaged?
If weed volume can't be started due to some damaged files，other object in the same volume can't be accessed,which is a big pitty

**Comment by chrislusf** [comment, marked as answer]
If .idx files are damaged, you can use `weed fix` to re-generate them.

If .dat files are damaged, and only the tail has problem, it should still be able to work.

**Comment by guol-fnst** [comment]
So if the whole .dat file was missing , the volume process couldn't work?

**Reply by chrislusf** [reply]
the volume server will create new volumes.
